src/main.c

```diff
@@ -486,10 +486,10 @@ int sub_08001D5C(u16 a, u16 b)
     return 16 * gUnk_03002970[a][b * 2 + 3];
 }
 
-int *sub_08001D78(int *a, u16 b)
+struct Unk_08001DE8_1 *sub_08001D78(int *a, u16 b)
 {
     int *var = &a[b * 2 + 2];
-    return &a[*var * 4];
+    return (struct Unk_08001DE8_1 *) &a[*var * 4]; // TODO: Fix cast.
```

Comment:
I can't think of how this could work. Maybe verify your type analysis again?

Reply:
I get this from the pseudo code of sub_8031420. It's something like this:    

v5 = sub_8001D78(v4, *((_BYTE *)v3 + 2));
sub_8005D6C(*((_BYTE *)v3 + 24), (Unk_08001DE8_1 *)v5);

I goto sub_8005D6C and get something like this: 

sub_8001DE8(a2, (Unk_08001DE8_2 *)&v10);

I don't know if the pseudo is wrong or something.